Add a text for displaying values on histogram.

lionelkusch opened the pull request:
For the histogram, I add text for displaying the percentage of each bar, the same as for values_count. I try to reuse the idea of display value for displaying on the histogram.

This pull request was made for fixing the issue #1454.

Vincent-Maladiere commented:
Hey @lionelkusch thank you for this PR! https://github.com/skrub-data/skrub/issues/1454 is kinda tricky and we're not sure how to move forward.

Your suggestion is interesting, but my understanding is that we'd rather have the height of each vertical bar rather than the percentile of their distribution. Additionally, we'd also like to have the total count for the categorical distribution (displayed horizontally). We need to see whether adding these figures would take too much space to be useful or not.

We can use this PR to explore different options.

lionelkusch commented:
This PR is quite limited to add information to the histogram is the same spirit as for the value_counted.
I pushed further in PR #1506 . For adding the height of the bar. 
Tell me what final form you want.

lionelkusch commented:
> > Addressing your remarks:

> > >   The problem when you combine the two in the same graphic is that the text will be longer than the columns. This won't look nice and it will be difficult to automatically add the text to the graphic.

> >  If the text crossing the bar edge doesn't look good, we can choose to always have the text outside of the bar, increasing the figure size slightly if necessary.

> > >  Adding the number of elements creates some variability and creates a difficult problem for the position of the text relative to the bar.

> > I think this can be addressed by controlling the number of digits in the format using custom rules, like full representation when the entries are smaller than 10,000 and scientific notation otherwise. We would have an upper bound on the size of the text.

> > -----------------------------------
> > What do the others think?

I agree on all points here and I would also prefer having both numbers visible at the same time. 

> Adding the number of elements creates some variability and creates a difficult problem for the position of the text relative to the bar.

How about adding the % to the right (the top) of the plot, aligned with each bar? With a fixed format (like 3 significant digits) the space should remain mostly constant, while the actual value is still represented in the same way as it is now. If the % sign takes too much space, it could be added above the column of numbers as an annotation.

_Originally posted by @rcap107 in https://github.com/skrub-data/skrub/issues/1506#issuecomment-3089379783_

I put back the comments of the PR #1506 about the graphic representation.

rcap107 commented:
Hey @lionelkusch, do you think you finish this PR? As we discussed, the current version is good, the only change would be removing the % from the numerical columns 

I hope there won't be too many conflicts to deal with

lionelkusch commented:
@rcap107  I am sorry, I forgot to do the modification.
I did it. I think that the PR is read.

rcap107 commented:
> Is there a place where I can look at a full HTML version, to interact a bit with it?

I was comparing the table report in the homepage of the generated docs with that in the live version

lionelkusch commented:
I removed the tick line; I think it's not needed it for the readability of the figure.

GaelVaroquaux commented:
It's quite OK, actually:

<img width="556" height="766" alt="image" src="https://github.com/user-attachments/assets/906db77a-e85a-424f-bb81-35304ff80bd9" />

My concern is: information overload. The more we put, the harder it is for a user to focus and to get the message. It brings us to the gold rule of communication: less in more.

But you've managed to keep it really light.

An ideal option, would have been to have a "hover": information that appears when the mouse goes over the relevant graph. But it's hard to do with our current toolkit.

I'm +1 for merge on this. Congratulations!!

rcap107 reviewed (approved):
Looks good to me, thanks a lot @lionelkusch. It was a bit convoluted (my fault for that), but the result is a nice addition to the TableReport.
